examples/keras_io/tensorflow/nlp/text_extraction_with_bert.py

```diff
@@ -0,0 +1,346 @@
+            tokenized_question.ids[1:]
+        )
+        attention_mask = [1] * len(input_ids)
+
+        # Pad and create attention masks.
+        # Skip if truncation is needed
+        padding_length = max_len - len(input_ids)
+        if padding_length > 0:  # pad
+            input_ids = input_ids + ([0] * padding_length)
+            attention_mask = attention_mask + ([0] * padding_length)
+            token_type_ids = token_type_ids + ([0] * padding_length)
+        elif padding_length < 0:  # skip
+            self.skip = True
+            return
+
+        self.input_ids = input_ids
+        self.token_type_ids = token_type_ids
+        self.attention_mask = attention_mask
+        self.start_token_idx = start_token_idx
+        self.end_token_idx = end_token_idx
+        self.context_token_to_char = tokenized_context.offsets
+
+
+with open(train_path) as f:
+    raw_train_data = json.load(f)
+
+with open(eval_path) as f:
+    raw_eval_data = json.load(f)
+
+
+def create_squad_examples(raw_data):
+    squad_examples = []
+    for item in raw_data["data"]:
+        for para in item["paragraphs"]:
+            context = para["context"]
+            for qa in para["qas"]:
+                question = qa["question"]
+                answer_text = qa["answers"][0]["text"]
+                all_answers = [_["text"] for _ in qa["answers"]]
+                start_char_idx = qa["answers"][0]["answer_start"]
+                squad_eg = SquadExample(
+                    question, context, start_char_idx, answer_text, all_answers
+                )
+                squad_eg.preprocess()
+                squad_examples.append(squad_eg)
+    return squad_examples
+
+
+def create_inputs_targets(squad_examples):
+    dataset_dict = {
+        "input_ids": [],
+        "token_type_ids": [],
+        "attention_mask": [],
+        "start_token_idx": [],
+        "end_token_idx": [],
+    }
+    for item in squad_examples:
+        if item.skip == False:
+            for key in dataset_dict:
+                dataset_dict[key].append(getattr(item, key))
+    for key in dataset_dict:
+        dataset_dict[key] = np.array(dataset_dict[key])
+
+    x = [
+        dataset_dict["input_ids"],
+        dataset_dict["token_type_ids"],
+        dataset_dict["attention_mask"],
+    ]
+    y = [dataset_dict["start_token_idx"], dataset_dict["end_token_idx"]]
+    return x, y
+
+
+train_squad_examples = create_squad_examples(raw_train_data)
+x_train, y_train = create_inputs_targets(train_squad_examples)
+print(f"{len(train_squad_examples)} training points created.")
+
+eval_squad_examples = create_squad_examples(raw_eval_data)
+x_eval, y_eval = create_inputs_targets(eval_squad_examples)
+print(f"{len(eval_squad_examples)} evaluation points created.")
+
+"""
+Create the Question-Answering Model using BERT and Functional API
+"""
+
+
+def create_model():
+    ## BERT encoder
+    encoder = TFBertModel.from_pretrained("bert-base-uncased")
```

Comment:
That's not going to work with Keras 3, is it? The Transformers library is still on Keras 2.

Reply:
The code runs fine with this but I get your point. So should we wait for the Transformers migration to Keras 3 then ? Or should I use BertModel instead of TFBertModel and adapt the example as a PyTorch implementation ?

Reply:
> The code runs fine with this

But did you install Keras 3? What's your Keras version?

Reply:
You're right ! I did install Keras 3 as per the instructions in a new environment but somewhat I had another version (Keras-nightly 2.15.0.dev2023092207) installed as well. The latter was the one being used ... I started from scratch and now indeed the transformers model does not work. I will close the PR. Sorry for the inconvenience.